lasagne/layers/recurrent.py

```diff
@@ -265,77 +414,13 @@ def __init__(self, input_layer, num_units,
-                :math:`W_{xi}`
-            - W_hid_to_ingate : function or np.ndarray or theano.shared
-                :math:`W_{hi}`
-            - W_cell_to_ingate : function or np.ndarray or theano.shared
-                :math:`W_{ci}`
-            - b_ingate : function or np.ndarray or theano.shared
-                :math:`b_i`
-            - nonlinearity_ingate : function
-                :math:`\sigma`
-            - W_in_to_forgetgate : function or np.ndarray or theano.shared
-                :math:`W_{xf}`
-            - W_hid_to_forgetgate : function or np.ndarray or theano.shared
-                :math:`W_{hf}`
-            - W_cell_to_forgetgate : function or np.ndarray or theano.shared
-                :math:`W_{cf}`
-            - b_forgetgate : function or np.ndarray or theano.shared
-                :math:`b_f`
-            - nonlinearity_forgetgate : function
-                :math:`\sigma`
-            - W_in_to_cell : function or np.ndarray or theano.shared
-                :math:`W_{ic}`
-            - W_hid_to_cell : function or np.ndarray or theano.shared
-                :math:`W_{hc}`
-            - b_cell : function or np.ndarray or theano.shared
-                :math:`b_c`
-            - nonlinearity_cell : function or np.ndarray or theano.shared
-                :math:`\tanh`
-            - W_in_to_outgate : function or np.ndarray or theano.shared
-                :math:`W_{io}`
-            - W_hid_to_outgate : function or np.ndarray or theano.shared
-                :math:`W_{ho}`
-            - W_cell_to_outgate : function or np.ndarray or theano.shared
-                :math:`W_{co}`
-            - b_outgate : function or np.ndarray or theano.shared
-                :math:`b_o`
-            - nonlinearity_outgate : function
-                :math:`\sigma`
-            - nonlinearity_out : function or np.ndarray or theano.shared
-                :math:`\tanh`
-            - cell_init : function or np.ndarray or theano.shared
-                :math:`c_0`
-            - hid_init : function or np.ndarray or theano.shared
-                :math:`h_0`
-            - backwards : boolean
-                If True, process the sequence backwards and then reverse the
-                output again such that the output from the layer is always
-                from x_1 to x_n.
-            - learn_init : boolean
-                If True, initial hidden values are learned
-            - peepholes : boolean
-                If True, the LSTM uses peephole connections.
-                When False, W_cell_to_ingate, W_cell_to_forgetgate and
-                W_cell_to_outgate are ignored.
-            - gradient_steps : int
-                Number of timesteps to include in backpropagated gradient
-                If -1, backpropagate through the entire sequence
-        '''
+                 gradient_steps=-1,
+                 grad_clipping=False):
+
+        # todo: add subnetwork option.
```

Comment:
I wanted to add an option for a subnetwork. The idea was that the hidden states from the lstm is passed through the subnetwork and used as additional input to the LSTM in the next timestep. I think it is commonly used in decoders. But we can remove it for now.